src/components/templates/discount-settings/index.tsx

```diff
@@ -0,0 +1,276 @@
+                    disabled={isEdit}
+                  />
+                </div>
+              )}
+            </Section>
+            <Section
+              title="Start date"
+              description="Schedule the discount to activate in the future."
+              tooltip="If you want to schedule the discount to activate in the future, you can set a start date here, otherwise the discount will be active immediately."
+            >
+              <div className="flex items-center gap-xsmall">
+                <DatePicker
+                  date={startDate}
+                  label="Start date"
+                  onSubmitDate={setStartDate}
+                />
+                <TimePicker
+                  label="Expiry time"
+                  date={startDate}
+                  onSubmitDate={setStartDate}
+                />
+              </div>
+            </Section>
+            <div>
+              <div className="flex items-center mb-2xsmall">
+                <Checkbox
+                  label="Discount has an expiry date?"
+                  checked={hasExpiryDate}
+                  onChange={() => {
+                    if (expiryDate) {
+                      setExpiryDate(undefined)
+                    } else {
+                      setExpiryDate(new Date())
+                    }
+                    setHasExpiryDate(!hasExpiryDate)
+                  }}
+                />
+                <div className="flex items-center ml-1.5">
+                  <InfoTooltip
+                    content={
+                      "If you want to schedule the discount to deactivate in the future, you can set an expiry date here."
+                    }
+                  />
+                </div>
+              </div>
+              {expiryDate && (
+                <>
+                  <p className="inter-small-regular text-grey-50 mb-base">
+                    Schedule the discount to deactivate in the future.
+                  </p>
+                  <div className="flex items-center gap-xsmall">
+                    <DatePicker
+                      label="Expiry date"
+                      date={expiryDate}
+                      onSubmitDate={setExpiryDate}
+                    />
+                    <TimePicker
+                      label="Expiry time"
+                      date={expiryDate}
+                      onSubmitDate={setExpiryDate}
+                    />
+                  </div>
+                </>
+              )}
+            </div>
+            {isDynamic && (
+              <AvailabilityDuration
+                value={availabilityDuration}
+                onChange={setAvailabilityDuration}
+              />
+            )}
+          </div>
+        )}
+      </ConnectForm>
+    </BodyCard>
+  )
+}
+
+type SectionProps = {
+  title: string
+  description: string
+  tooltip?: string
+}
+
+const Section: React.FC<SectionProps> = ({
+  title,
+  description,
+  tooltip,
+  children,
+}) => {
```

Comment:
Would it make sense to refactor this is a molecule?

Reply:
sure might be useful elsewhere 😄